pyodide_lock/spec.py

```diff
@@ -91,3 +101,151 @@ def check_wheel_filenames(self) -> None:
+            base_path = wheel_files[0].parent
+
+        target_python = version_parse(self.info.python)
+        python_binary_tag = f"cp{target_python.major}{target_python.minor}"
+        python_pure_tags = [
+            f"py2.py{target_python.major}",
+            f"py{target_python.major}",
+            f"py{target_python.major}{target_python.minor}",
+        ]
+
+        target_platform = self.info.platform + "_" + self.info.arch
+
+        new_packages = {}
+        new_package_wheels = {}
+        for f in wheel_files:
+            split_name = f.stem.split("-")
+            name = canonicalize_name(split_name[0])
+            version = split_name[1]
+            python_tag = split_name[-3]
+            split_name[-2]
+            platform_tag = split_name[-1]
+
+            if platform_tag == "any":
+                if python_tag not in python_pure_tags:
+                    raise RuntimeError(
+                        f"Wheel {f} is built for incorrect python version {python_tag},"
+                        f"this lockfile expects {python_binary_tag} "
+                        f"or one of {python_pure_tags}"
+                    )
+            elif platform_tag != target_platform:
+                raise RuntimeError(
+                    f"Wheel {f} is built for incorrect platform {platform_tag},"
+                    f"this lockfile expects {target_platform}"
+                )
+            else:
+                if python_tag != python_binary_tag:
+                    raise RuntimeError(
+                        f"Wheel {f} is built for incorrect python version {python_tag},"
+                        f" this lockfile expects {python_binary_tag}"
+                    )
+
+            file_name = base_url + str(f.relative_to(base_path))
+            install_dir = "site"
+            package_type = "package"
+            sha256 = _generate_package_hash(f)
+            imports = parse_top_level_import_name(f)
+
+            new_packages[name] = PackageSpec(
+                name=name,
+                version=version,
+                install_dir=install_dir,
+                file_name=file_name,
+                package_type=package_type,
+                sha256=sha256,
+                imports=imports,
+                depends=[],
+            )
+            new_package_wheels[name] = f
+
+        # now fix up the dependencies for each of our new packages
+        # n.b. this assumes existing packages have correct dependencies,
+        # which is probably a good assumption.
+
+        requirements_with_extras = []
+        for package in new_packages.values():
+            # add any requirements to the list of packages
+            our_depends = []
+            wheel_file = new_package_wheels[package.name]
+            requirements = get_wheel_dependencies(wheel_file, package.name)
+            for r in requirements:
+                req_marker = r.marker
+                req_name = canonicalize_name(r.name)
```

Comment:
Same question as in https://github.com/pyodide/pyodide-lock/pull/18/files#r1332985361  Are you sure the name should be normalized? Last I checked it wasn't in `pyodide-lock.json` and one install by package name usually not the normalized name. But I could be mistaked.